subsys/usb/usbc/usbc_stack.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT usbc_port
+
+#include <devicetree.h>
+#include <init.h>
+#include <smf.h>
+#include <usbc/usbc.h>
+#include <zephyr/kernel.h>
+#include <logging/log.h>
+
+#include "usbc_stack.h"
+
+#define LOG_MODULE_NAME usbc_stack
+LOG_MODULE_REGISTER(usbc_stack);
+
+static const struct usbc_subsystem_api usbc_api;
+static int usbc_subsys_init(const struct device *dev);
+
+#define THREAD_STACK(inst) K_THREAD_STACK_DEFINE(my_stack_area_##inst, CONFIG_USBC_STACK_SIZE);
+
+#define PORT_THREAD(inst) void run_usbc_##inst(void *port_dev, void *unused1, void *unused2) \
+	{										     \
+		const struct device *dev = (const struct device *)port_dev;		     \
+		struct usbc_port_data *port = dev->data;				     \
+		struct request_value *req;						     \
+		int32_t request;							     \
+											     \
+		while (1) {								     \
+			req = k_fifo_get(&port->request_fifo, K_NO_WAIT);		     \
+			request = (req != NULL) ? req->val : REQUEST_NOP;		     \
+			tmp_pe_run(dev, request);					     \
+			tmp_prl_run(dev);						     \
+			tmp_tc_run(dev, request);					     \
+			if (request == PRIV_PORT_REQUEST_SUSPEND) {			     \
+				k_thread_suspend(port->port_thread);			     \
+			}								     \
+			k_msleep(5);							     \
```

Comment:
magic sleep here?

Reply:
I changed the magic number to a Kconfig option